### PR TITLE
Unit test to find whether the number is a multiple of 3 or 5

### PR DESCRIPTION
**Pull request by Purvajagopalakrishnan**

### CODE REVIEW

**Comment by Harimaharajan** [comment]
It is preferred to have the solution should be directly under Repo, not inside a Folder

File: TDD_Training/XUnitTestProject/FindMultiples.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+using System;
+using Xunit;
+
+namespace XUnitTestProject
+{
+    public class UnitTest1
```
Comment (Harimaharajan) [inline TDD_Training/XUnitTestProject/FindMultiples.Tests.cs]: UnitTest1?

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+using System;
+using TDD_Training;
+using Xunit;
+
+namespace FindMultiplesTests
+{
+    public class FindMultiplesTests
+    {
+        private readonly FindMultiples findMultiplesService = new FindMultiples();
```
Comment (Harimaharajan) [inline TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: readonly fields naming convention 
_findMultiplesService

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        [Fact]
+        public void CheckGivenNumber_IfMultipleOfThree_ReturnsFizz()
```
Comment (Harimaharajan) [inline TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: Why not move this also to the below Theory Data?

File: TDD_Training/TDD_Training/FindMultiples.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace TDD_Training
+{
+    public class FindMultiples
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FindMultiples.cs]: MultiplicationService
or 
FizzBuzzService?

File: TDD_Training/TDD_Training/FindMultiples.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        public string FindMultiple(int givenNumber)
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FindMultiples.cs]: isMultipleOf()?

File: TDD_Training/TDD_Training/FindMultiples.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            try
+            {
+                if (givenNumber % 3 == 0 && givenNumber % 5 == 0)
+                {
+                    Console.WriteLine("FizzBuzz");
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FindMultiples.cs]: No need for Console.WriteLine's if you are returning the string

File: TDD_Training/TDD_Training/FindMultiples.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+                }
+                if (givenNumber % 5 == 0)
+                {
+                    Console.WriteLine("Buzz");
+                    return "Buzz";
+                }
+                Console.WriteLine(givenNumber);
+                return Convert.ToString(givenNumber);
+            }
+            catch (Exception ex)
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FindMultiples.cs]: You can remove the try-catch here I believe, if your logic won't throw any exception

File: TDD_Training/TDD_Training/Interface/Multiples.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TDD_Training.Interface
+{
+    public interface Multiples
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/Interface/Multiples.cs]: Is this being used?

File: TDD_Training/TDD_Training/Program.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            Console.WriteLine("Input number");
+            try
+            {
+                var inputNumber = int.Parse(Console.ReadLine());
+                FindMultiples findMultiples = new FindMultiples();
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/Program.cs]: You could've used the Interface if you have used it

File: TDD_Training/UnitTestProject/FindMultiples.Tests/FindMultiplesTests.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using Xunit;
+using System.Threading.Tasks;
+
+namespace UnitTestProject.FindMultiples.Tests
+{
+    /// <summary>
+    /// Summary description for FindMultiplesTests
+    /// </summary>
+    public class FindMultiplesTests
```
Comment (Harimaharajan) [inline TDD_Training/UnitTestProject/FindMultiples.Tests/FindMultiplesTests.cs]: I couldn't see the need for these test cases?
Can you explain??

**Review by Harimaharajan — CHANGES_REQUESTED** [review]
The PR has way too many unnecessary LOCs and files. 
Can you please, make sure to clean those up before raising a PR?

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs (hunk 2 above)
Comment (Purvajagopalakrishnan) [reply TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: Got it. I will update in the code

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs (hunk 3 above)
Comment (Purvajagopalakrishnan) [reply TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: I wrote one test based on [Fact] and rest on [theory]. Just to have an example for both

File: TDD_Training/TDD_Training/Interface/Multiples.cs (hunk 8 above)
Comment (Purvajagopalakrishnan) [reply TDD_Training/TDD_Training/Interface/Multiples.cs]: No. I thought of removing this before committing. But I forgot to remove it. I will update the code.

File: TDD_Training/UnitTestProject/FindMultiples.Tests/FindMultiplesTests.cs (hunk 10 above)
Comment (Purvajagopalakrishnan) [reply TDD_Training/UnitTestProject/FindMultiples.Tests/FindMultiplesTests.cs]: This is the exact folder where I wrote test cases. The above one (I mean the first PR comment) that file is not being used. I removed it actually. In my local, I don't find that above file. I will check one more time and push the changes.

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs (hunk 3 above)
Comment (Harimaharajan) [reply TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: Yep. That's good. But you can merge this along with Theory Data

File: TDD_Training/FindMultiplesTests/FindMultiplesTests.cs [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+            //Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData(6)]
+        [InlineData(93)]
+        [InlineData(108)]
+        public void CheckNumber_IfMultipleOfThree_ReturnsFizz(int value)
```
Comment (Harimaharajan) [inline TDD_Training/FindMultiplesTests/FindMultiplesTests.cs]: Method description should be declarative
something like
Given_when_then
GivenNumberInMultiplesOfThree_IsMultipleOf_ReturnsFizz()

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 12]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace TDD_Training
+{
+    public class FizzBuzzService
+    {
+        public string isMultipleOf(int givenNumber)
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: Ideally, this method is not checking for a multiple of correct?
it is returning the Fizz or Buzz based on your input.
So I'd rename something like
CheckFizzBuzz or something.

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            if (givenNumber % 3 == 0 && givenNumber % 5 == 0)
+            {
+                return "FizzBuzz";
+            }
+            if (givenNumber % 3 == 0)
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: Can you try moving this repetitive logic to a common method and reuse them?

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+            {
+                return "Fizz";
+            }
+            if (givenNumber % 5 == 0)
+            {
+                return "Buzz";
+            }
+            Console.WriteLine(givenNumber);
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: can remove the  cw.

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 15]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace TDD_Training
+{
+    public class FizzBuzzService
+    {
+        public string isMultipleOf(int givenNumber)
+        {
+            if (givenNumber % 3 == 0 && givenNumber % 5 == 0)
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: why not switch?

**Review by Harimaharajan — CHANGES_REQUESTED** [review]

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 16]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace TDD_Training
+{
+    public class FizzBuzzService
+    {
+        public string CheckFizzBuzz(int givenNumber)
+        {
+            switch(givenNumber % 3 == 0 && givenNumber % 5 == 0 ? "FizzBuzz" :
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: What's this??

Comment (Harimaharajan) [reply TDD_Training/TDD_Training/FizzBuzzService.cs]: If you are comparing everything as conditional statements, you won't even need switch. you can just return from there right??
This is not how switch is supposed to be used.

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+                }
+                default:
+                {
+                    return Convert.ToString(givenNumber);
+                }
+            }
+        }
+
+        public bool IsMultipleOfThree(int givenNumber)
+        {
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs:33]: You can do it in single line. the modulus check would by default gives you boolean. you dont have to return it. explicitly.
Can you change that?

File: TDD_Training/TDD_Training/FizzBuzzService.cs [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace TDD_Training
+{
+    public class FizzBuzzService
+    {
+        public string CheckFizzBuzz(int givenNumber)
+        {
+            switch(givenNumber)
+            {
+                case int _ when IsMultipleOfThreeOrFive(givenNumber) == true:
```
Comment (Harimaharajan) [inline TDD_Training/TDD_Training/FizzBuzzService.cs]: IsMultipleOfThree and IsMultipleOfFive

no need for `IsMultipleOfThreeOrFive`

**Review by Harimaharajan — APPROVED** [review]

**Pull request merged**